test/runner.test.ts

```diff
@@ -1,82 +1,44 @@
 import * as platformConstants from '../src/pure/platform'
 
-// Mock vscode ("pure" modules aren't quite pure)
-vi.mock('vscode', () => {
-  return {
-    default: { myDefaultKey: vi.fn() },
-    namedExport: vi.fn(),
-    window: {
-      createOutputChannel: () => {
-        return {
-          appendLine: vi.fn(),
-        }
-      },
-    },
-  }
-})
-
-// Mock config
-vi.mock('../src/config', () => {
-  return {
-    getConfig: () => {
-      return {
-        env: null,
-      }
-    },
-  }
-})
-
-// Mock runVitestWithApi, causing it to return its arguments as its output to allow us to assert their values
-vi.mock('../src/pure/ApiProcess', () => {
-  return {
-    runVitestWithApi: (
-      vitest: { cmd: string; args: string[] },
-      workspace: string,
-      handlers: any,
-      customStartProcess?: (config: any) => void,
-    ) => {
-      return `vitest.cmd=${vitest.cmd}`
-        + ` vitest.args=${vitest.args}`
-        + ` workspace=${workspace}`
-        + ` customStartProcess=${!!customStartProcess}`
-    },
-  }
-})
-
 describe('TestRunner', () => {
   const prevIsWindows = platformConstants.isWindows
 
   afterEach(() => {
     Object.defineProperty(platformConstants, 'isWindows', { value: prevIsWindows, writable: true })
   })
 
-  test.each([
-    [false, false, `vitest,abc.spec.ts,-t,a (b) "c" d`],
-    [false, true, `vitest,abc.spec.ts,-t,a (b) "c" d`],
-    [true, false, `vitest,abc.spec.ts,-t,"a (b) \\"c\\" d"`],
-    [true, true, `vitest,abc.spec.ts,-t,a (b) "c" d`],
-  ])('scheduleRun wrap test patterns if needed, (isWindows: %s, customStartProcess: %s)',
-    async (isWindows, useCustomStartProcess, expectedArgs) => {
-    Object.defineProperty(platformConstants, 'isWindows', { value: isWindows, writable: true })
-    const workspacePath = '/test'
-    const testFiles = ['abc.spec.ts']
-    const testNamePattern = 'a (b) "c" d'
-    const customStartProcess = useCustomStartProcess ? () => {} : undefined
-
-    const { testResultFiles, output } = await new TestRunner(workspacePath, undefined).scheduleRun(
-      testFiles,
-      testNamePattern,
-      undefined,
-      undefined,
-      undefined,
-      undefined,
-      undefined,
-      customStartProcess,
-    )
-
-    expect(testResultFiles).toBeDefined()
-    expect(output).toBe(`vitest.cmd=npx vitest.args=${expectedArgs} workspace=/test customStartProcess=${useCustomStartProcess}`)
+  const testCases = [
+    {isWindows: false, useCustomStartProcess: false, expectedArgs: `vitest,abc.spec.ts,-t,a (b) "c" d`},
+    {isWindows: false, useCustomStartProcess: true, expectedArgs: `vitest,abc.spec.ts,-t,a (b) "c" d`},
+    {isWindows: true, useCustomStartProcess: false, expectedArgs: `vitest,abc.spec.ts,-t,"a (b) \\"c\\" d"`},
+    {isWindows: true, useCustomStartProcess: true, expectedArgs: `vitest,abc.spec.ts,-t,a (b) "c" d`},
+  ]
+  testCases.forEach(function (testCase) {
+    const {isWindows, useCustomStartProcess, expectedArgs} = testCase
+    describe(
```

Comment:
I think these tests is not running because of `describe` instead of `it`. After replace it with `it`, the tests fail with:

```
  1) TestRunner
       scheduleRun wrap test patterns if needed, (isWindows: false, customStartProcess: false):
     TypeError: Cannot redefine property: isWindows
        at Function.defineProperty (<anonymous>)
        at Context.<anonymous> (test/runner.test.ts:2:1709)
        at process.processImmediate (node:internal/timers:476:21)

  2) TestRunner
       "after each" hook for "scheduleRun wrap test patterns if needed, (isWindows: false, customStartProcess: false)":
     TypeError: Cannot redefine property: isWindows
        at Function.defineProperty (<anonymous>)
        at Context.<anonymous> (test/runner.test.ts:2:989)
        at process.processImmediate (node:internal/timers:476:21)
```

Maybe, we actually need to mock something after all?

Reply:
Interesting. Although that failure doesn't seem to happen due to lack of mocking.

Reply:
I didn't realize at that time, but I think this test's purpose was not about "running" but only about cli argument sanitization as written on the left side of the diff:

> Mock runVitestWithApi, causing it to return its arguments as its output to allow us to assert their values

Probably the original author wanted to unit test code around here:

https://github.com/vitest-dev/vscode/blob/85aa7edfce24914bee6a849e61d935e8f0b97bb2/src/pure/runner.ts#L62-L96